store/store.go

```diff
@@ -175,12 +206,50 @@ func NewStore(
 		return nil, err
 	}
 
-	return &store{bqDataset, bqClient, mappings, outArcInfo,
-		inArcInfo, subTypeMap, containedIn, btClient.Open(btTable), cacheData}, nil
+	st := &store{bqDataset, bqClient, mappings, outArcInfo,
+		inArcInfo, subTypeMap, containedIn, btClient.Open(btTable), nil}
+
+	// Cloud PubSub receiver when branch cache is updated.
+	pubsubClient, err := pubsub.NewClient(ctx, pubsubProject)
+	if err != nil {
+		log.Fatalf("pubsub.NewClient: %v", err)
+	}
+	// Always create a new subscriber with default expiration date of 31 days.
+	subID := subIDPrefix + randomString()
+	sub, err := pubsubClient.CreateSubscription(ctx, subID,
+		pubsub.SubscriptionConfig{Topic: pubsubClient.Topic(pubsubTopic)})
+	if err != nil {
+		log.Fatalf("pubsub CreateSubscription: %v", err)
+	}
+	log.Printf("Subscriber ID: %s", subID)
+
+	// Start the receiver in a goroutine.
+	go func() {
+		err = sub.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
+			gcsFolder := string(msg.Data)
+			log.Printf("Got message: %q\n", string(gcsFolder))
+			msg.Ack()
+			err := st.LoadBranchCache(ctx, gcsFolder)
+			if err != nil {
+				log.Printf("Load cache data got error %s", err)
+			}
+		})
+		if err != nil {
+			log.Fatalf("Cloud pubsub receive: %v", err)
+		}
+	}()
+
+	// Initial branch cachel load.
+	err = st.LoadBranchCache(ctx, gcsFolder)
```

Comment:
What is gcsFolder at init time - the one set by flag? That could be obsolete. Should we do the "latest_branch_cache_version" file read?

Reply:
I will replace here by reading the  "latest_branch_cache_version" file when it's ready. This needs to be added in the import controller?

Reply:
Yes.